### PR TITLE
ZFIN-8418 refactor Expression Repository to be more compliant with Hi…

### PR DESCRIPTION
**Pull request by cmpich**
…bernate 5.x

### CODE REVIEW

File: source/org/zfin/expression/repository/HibernateExpressionRepository.java [hunk 1]
```diff
@@ -63,1372 +59,1296 @@
...
+
+	/**
+	 * Retrieve a single expression structure by ID.
+	 *
+	 * @param zdbID structure ID
+	 * @return expression structure
+	 */
+	public ExpressionStructure getExpressionStructure(String zdbID) {
+		Session session = HibernateUtil.currentSession();
+		return session.get(ExpressionStructure.class, zdbID);
```
Comment (rtaylorzfin) [inline source/org/zfin/expression/repository/HibernateExpressionRepository.java:1319]: Is there any potential for edge cases where ".get" returns null and ".uniqueResult" throws exception?

File: source/org/zfin/expression/repository/HibernateExpressionRepository.java [hunk 2]
```diff
@@ -1462,437 +1382,434 @@ public void deleteExpressionResultPerFigure(ExpressionResult2 result, Figure fig
...
+	 * Retrieve Expressions for a given term.
+	 *
+	 * @param term term
+	 * @return list of expressions
+	 */
+	@Override
+	public List<ExpressionResult> getExpressionsWithEntity(GenericTerm term) {
+		String hql = "select distinct expression from ExpressionResult expression where " +
+			"(entity.superterm = :term OR entity.subterm = :term) " +
+			" AND expressionFound = :expressionFound ";
```
Comment (rtaylorzfin) [inline source/org/zfin/expression/repository/HibernateExpressionRepository.java:1569]: It's personal preference, but I like to embed the "true" in the hql myself, so it reads "... AND expressionFound = true ..."

There are a few other places in the refactor that contain query.setParameter("someTerm", true) too.

File: source/org/zfin/expression/repository/HibernateExpressionRepository.java (hunk 1 above)
Comment (cmpich) [reply source/org/zfin/expression/repository/HibernateExpressionRepository.java:1319]: Not fully sure but:
returning a null should be ok. the calling method should handle the null.
and the exception being thrown when uniqueResult will retrieve multiple records is fine as is. If the result set should always be unique this exception will fastest show if it is not true in some cases. I definitely would not call uniqueResult() knowing that there are cases where it is not true. In that case, you should call list() and then check according to best business logic how to handle the violation. if you need to check if something has a unique resultset you better call list() and then return the boolean and then proceed with both logical pathways from there. You typically should not use exceptions to go one or the other way in your code.

File: source/org/zfin/expression/repository/HibernateExpressionRepository.java (hunk 2 above)
Comment (cmpich) [reply source/org/zfin/expression/repository/HibernateExpressionRepository.java:1569]: I agree! If a parameter is fixed / hard-coded then do it right in the hql. Only use the setParameter() method if it gets passed in and can have multiple values.

**Pull request merged**